unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs

```diff
@@ -0,0 +1,200 @@
+            }
+            else
+            {
+                RenderProfileManifest.i.currentProfile.Apply();
+            }
+
+            // Reset Object Update value without notifying
+            DataStore.i.skyboxConfig.objectUpdated.Set(false, false);
+        }
+
+        void Applyconfig()
+        {
+            SelectSkyboxConfiguration();
+
+            if (!configuration.useDirectionalLight)
+            {
+                directionalLight.gameObject.SetActive(false);
+            }
+            //DCL.DataStore.i.isProceduralSkyboxInUse.Set(true);
+
+            // Calculate time factor
+            if (minutesPerSecond <= 0)
+            {
+                minutesPerSecond = 0.01f;
+            }
+            timeNormalizationFactor = 60 / minutesPerSecond;
+        }
+
+        private void SelectSkyboxConfiguration()
+        {
+            string configToLoad = DEFAULT_SKYBOX_ID;
+
+            if (overrideDefaultSkybox)
+            {
+                configToLoad = overrideSkyboxID;
+                overrideDefaultSkybox = false;
+            }
+            configuration = Resources.Load<SkyboxConfiguration>("Skybox Configurations/" + configToLoad);
+
+            if (configuration == null)
+            {
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
+                Debug.LogError("No configuration found in Resources. (Default path through tool is Assets/Scripts/Resources/Skybox Configurations)");
+#endif
+                return;
+            }
+
+            // Apply material as per number of Slots.
+            //TODO: Change shader on same material instead of having multiple material.
+            MaterialReferenceContainer.Mat_Layer matLayer = MaterialReferenceContainer.i.GetMat_LayerForLayers(configuration.slots.Count);
+            if (matLayer == null)
+            {
+                matLayer = MaterialReferenceContainer.i.materials[0];
+            }
+
+            configuration.ResetMaterial(matLayer.material, matLayer.numberOfSlots);
+            selectedMat = matLayer.material;
+
+            if (DataStore.i.skyboxConfig.useProceduralSkybox.Get())
+            {
+                RenderSettings.skybox = selectedMat;
+            }
+        }
+
+        // Update is called once per frame
+        public override void Update()
+        {
+            if (configuration == null || isPaused || !DataStore.i.skyboxConfig.useProceduralSkybox.Get())
+            {
+                return;
+            }
+            timeOfTheDay += Time.deltaTime / timeNormalizationFactor;
+            timeOfTheDay = Mathf.Clamp(timeOfTheDay, 0.01f, cycleTime);
+
+            configuration.ApplyOnMaterial(selectedMat, timeOfTheDay, GetNormalizedDayTime(), directionalLight);
+
+            if (timeOfTheDay >= cycleTime)
+            {
+                timeOfTheDay = 0.01f;
+            }
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+            //DCL.DataStore.i.isProceduralSkyboxInUse.Set(false);
```

Comment:
```suggestion
```

Reply:
Done. Making all these changes in a new commit